### PR TITLE
Implement test and fix pack-related intrinsics

### PR DESCRIPTION
**Pull request by marktwtn**
- test: Add tests for _mm_pack[u]s_epi{16,32}
- fix: Fix conversion of _mm_packus_epi32

### CODE REVIEW

File: tests/impl.cpp [hunk 1]
```diff
@@ -5207,7 +5290,34 @@ result_t test_mm_mullo_epi32(const SSE2NEONTestImpl &impl, uint32_t i)
 
 result_t test_mm_packus_epi32(const SSE2NEONTestImpl &impl, uint32_t i)
 {
-    return TEST_UNIMPL;
+    uint16_t max = 65535;
```
Comment (jserv) [inline tests/impl.cpp:5293]: How about specifying `UINT16_MAX`?

File: tests/impl.cpp [hunk 2]
```diff
@@ -3226,17 +3226,100 @@ result_t test_mm_or_si128(const SSE2NEONTestImpl &impl, uint32_t i)
...
+    __m128i b = do_mm_load_ps((const int32_t *) _b);
+    __m128i c = _mm_packs_epi32(a, b);
+
+    return validateInt16(c, d[0], d[1], d[2], d[3], d[4], d[5], d[6], d[7]);
 }
 
 result_t test_mm_packus_epi16(const SSE2NEONTestImpl &impl, uint32_t i)
 {
-    return TEST_UNIMPL;
+    uint8_t max = 255;
```
Comment (jserv) [inline tests/impl.cpp:3294]: How about specifying `UINT8_MAX`?

File: tests/impl.cpp [hunk 3]
```diff
@@ -3226,17 +3226,100 @@ result_t test_mm_or_si128(const SSE2NEONTestImpl &impl, uint32_t i)
...
+    __m128i c = _mm_packs_epi16(a, b);
+
+    return validateInt8(c, d[0], d[1], d[2], d[3], d[4], d[5], d[6], d[7], d[8],
+                        d[9], d[10], d[11], d[12], d[13], d[14], d[15]);
 }
 
 result_t test_mm_packs_epi32(const SSE2NEONTestImpl &impl, uint32_t i)
 {
-    return TEST_UNIMPL;
+    int16_t max = 32767;
```
Comment (jserv) [inline tests/impl.cpp:3262]: How about `INT8_MAX` and `INT8_MIN`?

File: tests/impl.cpp (hunk 1 above)
Comment (marktwtn) [reply tests/impl.cpp]: Done.

File: tests/impl.cpp (hunk 3 above)
Comment (marktwtn) [reply tests/impl.cpp]: Done.

File: tests/impl.cpp (hunk 2 above)
Comment (marktwtn) [reply tests/impl.cpp]: Done.

**Pull request merged**